aeppl/transforms.py

```diff
@@ -66,76 +34,231 @@ def log_jac_det(self, value: TensorVariable, *inputs) -> TensorVariable:
+class TransformedRVMeta(MetaType):
+    def __new__(cls, name, bases, clsdict):
+        cls_res = super().__new__(cls, name, bases, clsdict)
+        base_op = clsdict.get("base_op", None)
+        default = clsdict.get("default", False)
+
+        if default and base_op is not None:
+            # Create dispatch functions
+            @_get_default_transformed_rv_op.register(type(base_op))
+            def get_default_transformed_rv_op(op):
+                return cls_res()
+
+        return cls_res
+
+
+class TransformedRV(RandomVariable, metaclass=TransformedRVMeta):
     r"""A base class for transformed `RandomVariable`\s."""
 
 
-@_logprob.register(TransformedRV)
-def transformed_logprob(op, value, *inputs, name=None, **kwargs):
+@singledispatch
+def _get_default_transformed_rv_op(
+    op: Op,
+) -> Optional[Type[TransformedRV]]:
+    """Return a default ``TransformedRV`` for a given ``RandomVariable`` ``Op``.
+
+    This function dispatches on the type of ``Op``, which should be a subclass
+    of ``RandomVariable``.  If you want to implement new transforms for a
+    ``RandomVariable``, register a function on this dispatcher.
+
     """
-    Compute logp graph for a value variable that was back-transformed to be on
-    the natural support of the respective random variable.
+    return None
+
+
+def create_transformed_rv_op(
+    rv_op: Op,
+    transform: Transform,
+    *,
+    default: bool = False,
+    cls_dict_extra: Optional[Dict] = None,
+) -> Type[TransformedRV]:
+    """Create a new ``TransformedRV`` given a base ``RandomVariable`` ``Op``
+
+    Pass `default = True` to specify that the returned ``TransformedRV`` should
+    be the default transformation for the base ``RandomVariable``
+
     """
 
-    logprob = _logprob(op.base_op, value, *inputs, name=name, **kwargs)
+    trans_name = getattr(transform, "name", "transformed")
+    rv_type_name = type(rv_op).__name__
+    cls_dict = type(rv_op).__dict__.copy()
+    rv_name = cls_dict.get("name", "")
+    if rv_name:
+        cls_dict["name"] = f"{rv_name}_{trans_name}"
+    cls_dict["base_op"] = rv_op
+    cls_dict["transform"] = transform
+    cls_dict["default"] = default
 
-    original_forward_value = op.transform.forward(value, *inputs)
-    jacobian = op.transform.log_jac_det(original_forward_value, *inputs)
+    if cls_dict_extra is not None:
+        cls_dict.update(cls_dict_extra)
 
-    if name:
-        logprob.name = f"{name}_logprob"
-        jacobian.name = f"{name}_logprob_jac"
+    new_op_type = type(f"Transformed{rv_type_name}", (TransformedRV,), cls_dict)
 
-    return logprob + jacobian
+    return new_op_type
+
+
+# =============
+# Transform Opt
+# =============
+
+
+DEFAULT_TRANSFORM = object()
+
+
+class TransformValuesMapping(Feature):
+    r"""Store dictionary indicating the transforms to apply to each value variable"""
+
+    def __init__(self, transform_values):
+        self.transform_values = transform_values
```

Comment:
It would make more sense to call this variable something like `values_to_transforms`, or `vars_to_transforms`.  The current name implies somethings else.

Reply:
I like `values_to_transforms`